experimental/devices/ccs811/ccs811.go

```diff
@@ -0,0 +1,361 @@
+package ccs811
+
+import (
+	"fmt"
+	"time"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/i2c"
+)
+
+// Different measurement modes constants
+const (
+	MeasurementModeIdle         byte = 0
+	MeasurementModeConstant1000 byte = 1
+	MeasurementModePulse        byte = 2
+	MeasurementModeLowPower     byte = 3
+	MeasurementModeConstant250  byte = 4
+)
+
+// NeededData represents set of data read from the sensor
+type NeededData byte
+
+// What data should be read from the sensor
+const (
+	ReadCO2          NeededData = 2
+	ReadCO2VOC       NeededData = 4
+	ReadCO2VOCStatus NeededData = 5
+	ReadAll          NeededData = 8
+)
+
+// SensorErrorID represents error reported by the sensor
+type SensorErrorID byte
+
+/*
+Error constants, applicable if status registers signals error
+0
+WRITE_REG_INVALID
+The CCS811 received an I2C write request addressed to this station
+but with invalid register address ID
+1
+READ_REG_INVALID
+The CCS811 received an I2C read request to a mailbox ID that is invalid
+2
+MEASMODE_INVALID
+The CCS811 received an I2C request to write an unsupported mode to MEAS_MODE
+3
+MAX_RESISTANCE
+The sensor resistance measurement has reached or exceeded the maximum range
+4
+HEATER_FAULT
+The Heater current in the CCS811 is not in range
+5
+HEATER_SUPPLY
+The Heater voltage is not being applied correctly
+*/
+const (
+	WriteRegInvalid SensorErrorID = 0x1
+	ReadRegInvalid  SensorErrorID = 0x2
+	MeasModeInvalid SensorErrorID = 0x4
+	MaxResistance   SensorErrorID = 0x8
+	HeaterFault     SensorErrorID = 0x10
+	HeaterSupply    SensorErrorID = 0x20
+)
+
+// Opts holds the configuration options.
+type Opts struct {
+	Addr               uint16
+	MeasurementMode    byte
+	InterruptWhenReady bool
+	UseThreshold       bool
+}
+
+// DefaultOpts are the safe default options.
+var DefaultOpts = Opts{
+	Addr:               0x5A,
+	MeasurementMode:    MeasurementModeConstant1000,
+	InterruptWhenReady: false,
+	UseThreshold:       false,
+}
+
+// New creates a new driver for CCS811 VOC sensor
+func New(bus i2c.Bus, opts *Opts) (*Dev, error) {
+	if opts.Addr != 0x5A && opts.Addr != 0x5B {
+		return nil, fmt.Errorf("Invalid device address, only 0x5A or 0x5B are allowed")
+	}
+
+	if opts.MeasurementMode > 4 {
+		return nil, fmt.Errorf("Invalid measurement mode")
+	}
+
+	dev := &Dev{
+		c:    &i2c.Dev{Bus: bus, Addr: opts.Addr},
+		opts: opts,
```

Comment:
That's dangerous, the user could keep a pointer to the opts and change it without your knowledge. Make line 112 a value, not a pointer.